### PR TITLE
[release/7.0] Update gRPC and Protobuf dependencies

### PR DESCRIPTION
**Pull request by JamesNK**
# Update gRPC and Protobuf dependencies

Update gRPC and Protobuf dependencies in gRPC project template

## Description

* Update gRPC package versions to latest
* Update Protobuf package version to latest
* Remove warning suppressions for generated code (see issue)

Updates gRPC and Protobuf versions that are referenced by the gRPC template. These versions include a net7.0 target and fix warnings from generated code.

Fixes https://github.com/aspnet/AspNetCore-ManualTests/issues/1357

## Customer Impact

gRPC template shouldn't generate warnings by default.

## Regression?

- [x] Yes
- [ ] No

.NET 7 adds an analyzer that causes warnings in generated code

## Risk

- [ ] High
- [ ] Medium
- [x] Low

Just updating versions in template

## Verification

- [x] Manual (required)
- [x] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [x] N/A

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -245,12 +245,12 @@
...
-    <GrpcAspNetCoreServerVersion>2.45.0</GrpcAspNetCoreServerVersion>
-    <GrpcAuthVersion>2.45.0</GrpcAuthVersion>
-    <GrpcNetClientVersion>2.45.0</GrpcNetClientVersion>
-    <GrpcToolsVersion>2.45.0</GrpcToolsVersion>
+    <GoogleProtobufVersion>3.21.5</GoogleProtobufVersion>
+    <GrpcAspNetCoreVersion>2.49.0-pre1</GrpcAspNetCoreVersion>
+    <GrpcAspNetCoreServerVersion>2.49.0-pre1</GrpcAspNetCoreServerVersion>
+    <GrpcAuthVersion>2.49.0-pre1</GrpcAuthVersion>
+    <GrpcNetClientVersion>2.49.0-pre1</GrpcNetClientVersion>
+    <GrpcToolsVersion>2.49.0-pre1</GrpcToolsVersion>
```
Comment (dougbu) [inline eng/Versions.props:253]: We can't move to prerelease versions w/o messing up the RTM release. Are you sure these will have stable versions before we ship❔

File: src/Grpc/Interop/test/testassets/InteropClient/InteropClient.csproj [hunk 2]
```diff
@@ -4,8 +4,6 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>$(DefaultNetCoreTargetFramework)</TargetFramework>
     <Nullable>enable</Nullable>
-    <!-- Suppress warnings about lowercase variable names in generated code -->
```
Comment (dougbu) [inline src/Grpc/Interop/test/testassets/InteropClient/InteropClient.csproj:7]: This feels like a regression. Why are the analyzers complaining about generated code at all❔

File: eng/Versions.props (hunk 1 above)
Comment (JamesNK) [reply eng/Versions.props:253]: Yes. Should be available in 1-2 weeks.

Worst case, we revert.

File: src/Grpc/Interop/test/testassets/InteropClient/InteropClient.csproj (hunk 2 above)
Comment (JamesNK) [reply src/Grpc/Interop/test/testassets/InteropClient/InteropClient.csproj:7]: https://github.com/aspnet/AspNetCore-ManualTests/issues/1357

**Review by wtgodbe — APPROVED** [review]

File: eng/Versions.props [hunk 3]
```diff
@@ -245,12 +245,12 @@
...
     <FSharpCoreVersion>6.0.0</FSharpCoreVersion>
     <GoogleApiCommonProtosVersion>2.5.0</GoogleApiCommonProtosVersion>
-    <GoogleProtobufVersion>3.18.1</GoogleProtobufVersion>
-    <GrpcAspNetCoreVersion>2.45.0</GrpcAspNetCoreVersion>
-    <GrpcAspNetCoreServerVersion>2.45.0</GrpcAspNetCoreServerVersion>
-    <GrpcAuthVersion>2.45.0</GrpcAuthVersion>
-    <GrpcNetClientVersion>2.45.0</GrpcNetClientVersion>
-    <GrpcToolsVersion>2.45.0</GrpcToolsVersion>
+    <GoogleProtobufVersion>3.21.5</GoogleProtobufVersion>
+    <GrpcAspNetCoreVersion>2.49.0-pre1</GrpcAspNetCoreVersion>
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply eng/Versions.props:249]: Stable dependencies are currently blocked on grpc/grpc releasing a stable `Grpc.Tools` package. It's taking longer to release than expected.

**Comment by JamesNK** [comment]
Will wait for stable dependencies before merging. Currently waiting on https://github.com/grpc/grpc to release stable 2.49.0 of `Grpc.Tools`.

**Comment by JamesNK** [comment]
Updated to 2.49.0. Ready to merge.

**Comment by danmoseley** [comment]
Approved - we need to ship with stable dependencies, and with samples matching.

**Comment by JamesNK** [comment]
@dotnet/aspnet-admins Please merge

**Pull request merged**